CMakeLists.txt

```diff
@@ -258,7 +260,13 @@ IF(WITH_CORE)
   ENDIF(WITH_QTWEBKIT)
   #############################################################
   # search for Qt5
-  SET(QT_MIN_VERSION 5.2.0)
+  IF (WITH_3D)
+    # for 3D support we strictly require Qt >= 5.8
+    # (Qt 3D was introduced in 5.7 but it is not stable enough in that branch)
+    SET(QT_MIN_VERSION 5.8.0)
```

Comment:
Hi @wonder-sk 

I'm running 5.8 for now and I'm having trouble to compile master. I guess we should update to 5.9 the minimum version.

In 5.8:
```
/Users/etienne/dev/cpp/QGIS/src/3d/terrain/qgsterraintileloader_p.cpp:77: error: no type named 'QTextureMaterial' in namespace 'Qt3DExtras'
  Qt3DExtras::QTextureMaterial *material;
  ~~~~~~~~~~~~^
```

I also started this patch:
```
etienne@:~/dev/cpp/QGIS (master *) $ git diff
diff --git a/src/3d/qgs3dmapscene.cpp b/src/3d/qgs3dmapscene.cpp
index 3c1b47f867..db13fa4c16 100644
--- a/src/3d/qgs3dmapscene.cpp
+++ b/src/3d/qgs3dmapscene.cpp
@@ -153,7 +153,9 @@ Qgs3DMapScene::Qgs3DMapScene( const Qgs3DMapSettings &map, Qt3DExtras::QForwardR
     // docs say frustum culling must be disabled for skybox.
     // it _somehow_ works even when frustum culling is enabled with some camera positions,
     // but then when zoomed in more it would disappear - so let's keep frustum culling disabled
-    defaultFrameGraph->setFrustumCullingEnabled( false );
+#if QT_VERSION >= 0x050900
+  defaultFrameGraph->setFrustumCullingEnabled( false );
+#endif
   }
 
   // force initial update of chunked entities
```

Or is-it supposed to work on 5.8 too?

Reply:
Thanks for the report. I guess we could raise the minimum version to 5.9 since that is the release with long-term support, so I don't think distributions would stick to 5.8.

QTextureMaterial could be replaced by QDiffuseMapMaterial that is available since 5.7 (texture material is just a simplified material)

For frustum culling, that property indeed does not seem to exist in < 5.9 even though docs do not state that...

Reply:
I guess I will run into some other issues after I fix these two errors. So let's push the QT minimum version to 5.9 instead.

 Thanks Martin ;-)